server/src/project/project.service.ts

```diff
@@ -283,8 +284,50 @@ export class ProjectService {
   constructor(
     private readonly dynamicsWebApi: OdataService,
     private readonly config: ConfigService,
+    private readonly carto: CartoService,
   ) {}
 
+  async getBblsGeoJSON(bbls) {
+    let collectedBBLs = bbls;
+
+    if (collectedBBLs === null) return null;
+
+    collectedBBLs = collectedBBLs.filter(bbl => bbl !== null);
+
+    const SQL = `
+        SELECT
+          ST_AsGeoJSON(ST_Multi(ST_Union(the_geom))) AS polygons
+        FROM mappluto
+        WHERE bbl IN (${collectedBBLs.join(',')})
+        GROUP BY version
+      `;
+
+    const cartoResponse = await this.carto.fetchCarto(SQL, 'json', 'post');
+
+    if (cartoResponse.length === 0) {
+      return {
+        polygons: null,
+      };
+    }
+
+    // return first object in carto response, carto.sql always return an array
+    return cartoResponse[0];
+  }
+
+  async getBblsFeaturecollection(bbls) {
+    const { polygons: bblsGeoJSON } = await this.getBblsGeoJSON(bbls);
+
+    return {
+      type: 'FeatureCollection',
+      features: [
+        {
+          type: 'Feature',
+          geometry: JSON.parse(bblsGeoJSON),
```

Comment:
Wait, this has to be parsed to JSON? This should definitely happen in the carto service, no?

Reply:
Hmm, yeah maybe the Carto service can provide a utilitly function for this transformation. I supposed that transforming into a FeatureCollection is not necessarily carto-related, per se. But it would be nice to encapsulate that so consumers have less legwork and details to worry about..